docs/developing-fuzzbench/adding_a_new_benchmark.md

```diff
@@ -35,22 +35,57 @@ powerful but also more work.
 
 ## OSS-Fuzz benchmarks
 
-You can use any existing OSS-Fuzz project as a benchmark. You just need to
-create a single file `oss-fuzz.yaml` in the benchmark directory.
+You can use most existing OSS-Fuzz projects a benchmark. First find out which
+project and the fuzz target you want to use as a benchmark. Then find out the commit 
+from the project repo which you want to use. Finally, find out the date and time 
+(UTC) of that commit in iso format. This can be done in the project repo as follows:
+```shell
+git --no-pager log -1 $COMMIT_HASH --format=%cd --date=iso-strict
+```
 
-```yaml
-project: <project-name>
-fuzz_target: <fuzz-target-name>
-oss_fuzz_builder_hash: <sha>
+Once you have this information, run
+`benchmarks/oss_fuzz_benchmark_integration.py` to copy the necessary integration
+files, like so:
+
+```shell
+PYTHONPATH=. python3 benchmarks/oss_fuzz_benchmark_integration.py -p $PROJECT
+    -f $FUZZ_TARGET -c $COMMIT_HASH -d $COMMIT_DATE
 ```
-* `project` should be a [valid OSS-Fuzz project](https://github.com/google/oss-fuzz/tree/master/projects).
-* `fuzz_target` should be the name of a binary fuzz target from the project that we want to fuzz as the benchmark.
-* `oss_fuzz_builder_hash` is a SHA256 hash of a docker image from
-[gcr.io/fuzzbench/oss-fuzz](https://console.cloud.google.com/gcr/images/fuzzbench/GLOBAL/oss-fuzz?gcrImageListsize=30).
-Unless you are a project maintainer, you cannot build these images yourself.
-Therefore, let us know the benchmark you want incorporated and we will produce a
-build that can be used here.
-Example: [wireshark_fuzzshark_ip](https://github.com/google/fuzzbench/blob/master/benchmarks/wireshark_fuzzshark_ip/oss-fuzz.yaml).
+Example :
+```shell
+PYTHONPATH=. python3 benchmarks/oss_fuzz_benchmark_integration.py -p bloaty
+    -f fuzz_target -c f572d396fae9206628714fb2ce00f72e94f2258f -d 2019-10-19T09:07:25+01:00
+```
+
+The script should create the benchmark directory in
+`benchmarks/$PROJECT_$FUZZ_TARGET` (unless you specify the name manually) with
+all the files needed to build the benchmark. You must then remove unecessary files
+such as fuzz targets which are not used for the benchmark. Further, the `build.sh`
+file must be modified accordingly so as to only build the required fuzz target.
+
+Add the files in the directory to git (and then commit them):
+
+```shell
+git add benchmarks/$BENCHMARK/*
+```
+
+Test your integration:
+
+```shell
+export FUZZER_NAME=afl
+export BENCHMARK_NAME=zlib_zlib_uncompress_fuzzer
+
+make build-$FUZZER_NAME-$BENCHMARK_NAME
+make run-$FUZZER_NAME-$BENCHMARK_NAME
+```
+
+Add your benchmark to the list of OSS-Fuzz benchmarks in
+[test_fuzzer_benchmarks.py](https://github.com/google/fuzzbench/blob/master/.github/workflows/test_fuzzer_benchmarks.py)
```

Comment:
`[.github/workflows/build_and_test_run_fuzzer_benchmarks.py](https://github.com/google/fuzzbench/blob/master/.github/workflows/build_and_test_run_fuzzer_benchmarks.py)`.
Can you add a TODO to that file to get the list of benchmarks automatically?